composer/utils/object_store/oci_object_store.py

```diff
@@ -132,17 +145,41 @@ def download_object(
             os.makedirs(dirname, exist_ok=True)
         tmp_path = str(filename) + f'.{uuid.uuid4()}.tmp'
 
+        # Get the size of the object
+        head_object_response = self.client.head_object(self.namespace, self.bucket, object_name)
+        object_size = head_object_response.headers['content-length']
+        # Calculate the part sizes
+        base_part_size = int(object_size) // num_parts
+        remainder = int(object_size) % num_parts
+        part_sizes = [base_part_size] * num_parts
+        for i in range(remainder):
+            part_sizes[i] += 1
+        part_sizes = [part_size for part_size in part_sizes if part_size > 0]
+
```

Comment:
I'm worried about the temporary files not being cleaned up. The better way to make sure this is not the issue is to use a 
```with tempfile.TemporaryDirectory(dir=dirname) as temp_dir:
```
When ever this context manager is exited, all files in the folder will be deleted regardless of whether it exists because of an error etc.
`dir=dirname` ensures that the files are written to same device / location as the target file, but in a subfolder. You may also want to look at the prefix/suffix arg for additional customization.